src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const peopleArray = people.filter(person => century
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+    : person.sex === 'm');
+
+  const sumAgePeople = peopleArray
+    .reduce((sumAge, person) => sumAge + (person.died - person.born), 0);
+
+  const averageAgePeople = sumAgePeople / peopleArray.length;
```

Comment:
move this to the separate function & reuse it
<img width="639" alt="image" src="https://user-images.githubusercontent.com/91411692/227925718-f4127a65-8a5a-4e9f-9370-891eb0a3910b.png">